### PR TITLE
Add automated pull request creation.

### PR DESCRIPTION
**Pull request by brendandburns**
@thockin @zmerlynn

### CODE REVIEW

**Comment by zmerlynn** [comment]
Heh. This is the exact tool I was going to use to automate this, but I debated whether to require the dep.

**Comment by zmerlynn** [comment]
Can you show a sample? I haven't seen you open a successful one yet. :>

**Comment by brendandburns** [comment]
Fixed.  Successful PR is here:

https://github.com/GoogleCloudPlatform/kubernetes/pull/12246

File: hack/cherry_pick_pull.sh [hunk 1]
```diff
@@ -124,17 +133,9 @@ done
...
-  echo "    presumably ${BRANCH}. (This selection is near the top left.)"
-  echo
-  echo "    Use this subject: 'Automated cherry pick of ${PULLSUBJ}' and include a justification."
-  echo
-  echo "    Note: the tools actually scrape the branch name you just pushed, so don't worry about"
-  echo "    the subject too much, but DO keep at least ${NEWBRANCHREQ} in the remote branch name."
-  echo
+  echo "+++ Creating a pull request on github"
+  echo hub pull-request -m "Automated cherry pick of ${PULLSUBJ}" -h ${GITHUB_USER}:${NEWBRANCH} -b GoogleCloudPlatform:`basename ${BRANCH}`
+  hub pull-request -m "Automated cherry pick of ${PULLSUBJ}" -h ${GITHUB_USER}:${NEWBRANCH} -b GoogleCloudPlatform:`basename ${BRANCH}`
```
Comment (zmerlynn) [inline hack/cherry_pick_pull.sh]: nit: $() vs ``

Comment (zmerlynn) [inline hack/cherry_pick_pull.sh]: And forgot nit: quotes for "${GITHUB_USER}:${NEWBRANCH}" and "${BRANCH}"

Comment (brendandburns) [inline hack/cherry_pick_pull.sh]: done.

**Comment by brendandburns** [comment]
Nits addressed, ptal.

thanks
--brendan

File: hack/cherry_pick_pull.sh (hunk 1 above)
Comment (zmerlynn) [inline hack/cherry_pick_pull.sh]: Do we have somewhere to dump a link to https://google-styleguide.googlecode.com/svn/trunk/shell.xml ? Maybe should just have a style guide landing page for k8s.

(In the case of bash, we've settled on basically that style guide with the exception that we strongly prefer `.sh` over naked script names.)

**Comment by zmerlynn** [comment]
LGTM

**Pull request merged**